src/Kinto.elm

```diff
@@ -201,44 +199,50 @@ configure baseUrl auth =
 
 bodyDataDecoder : Json.Decoder Json.Value
 bodyDataDecoder =
-    ("data" := Json.value)
+    (field "data" Json.value)
 
 
 errorDecoder : Json.Decoder ErrorRecord
 errorDecoder =
-    Json.object4 ErrorRecord
-        ("errno" := Json.int)
-        ("message" := Json.string)
-        ("code" := Json.int)
-        ("error" := Json.string)
-
-
-toKintoResponse : Http.Response -> Result Error Json.Value
-toKintoResponse ({ value, status, statusText } as response) =
-    (extractBody response)
-        `Result.andThen`
-            (extractData
-                >> Result.formatError (extractError status statusText)
-            )
-
-
-extractBody : Http.Response -> Result Error String
-extractBody { value, status, statusText } =
-    case value of
-        Http.Text body ->
-            Ok body
-
-        _ ->
-            Err (ServerError status statusText "Unsupported response body")
-
-
-extractData : String -> Result String Json.Value
-extractData data =
-    Json.decodeString bodyDataDecoder data
-
-
-extractError : StatusCode -> StatusMsg -> String -> Error
-extractError statusCode statusMsg body =
+    Json.map4 ErrorRecord
+        (field "errno" Json.int)
+        (field "message" Json.string)
+        (field "code" Json.int)
+        (field "error" Json.string)
+
+
+toKintoResponse :
+    (Result Error Json.Value -> msg)
+    -> Result Http.Error Json.Value
+    -> msg
+toKintoResponse toMsg response =
+    response
+        |> Result.mapError extractError
+        |> toMsg
```

Comment:
What's the value of having `toMsg` as a parameter instead of just calling it afterwards?